File: ruby/sum_of_array_of_integers.rb
# write a function that takes an array of integers and returns the sum of all elements

array = [1, 2, 3, 4, 5] # -> 15

def add_array(arr)
  return arr.sum
end

# # without using Ruby method
# def add_array(arr)
#   total = 0
#   arr.each do |num|
#     total += num
#   end
#   return total
# end

p add_array(array)

File: javascript/sum_of_array_of_integers.js
// write a function that takes an array of integers and returns the sum of all elements

function addArray(arr) {
    var total = 0;
    for (var i = 0; i < arr.length; i++) {
        total += arr[i];
    }
    
    return total;
}

console.log(addArray([1, 2, 3, 4, 5])); // -> 15